Documenting building with simd feature

**Describe the bug**
```
~/release/apache-arrow-datafusion-5.0.0 via 🦀 v1.54.0 
✦ ❯ cargo build --release --features "simd"
   Compiling packed_simd_2 v0.3.5
error[E0554]: `#![feature]` may not be used on the stable release channel
   --> /home/andy/.cargo/registry/src/github.com-1ecc6299db9ec823/packed_simd_2-0.3.5/src/lib.rs:214:1
    |
214 | / #![feature(
215 | |     const_generics,
216 | |     repr_simd,
217 | |     rustc_attrs,
...   |
227 | |     llvm_asm
228 | | )]
    | |__^

```

**To Reproduce**
Run command above.

--- a/docs/user-guide/src/library.md
+++ b/docs/user-guide/src/library.md
@@ -19,6 +19,8 @@
 
 # Using DataFusion as a library
 
+## Default Configuration
+
 DataFusion is [published on crates.io](https://crates.io/crates/datafusion), and is [well documented on docs.rs](https://docs.rs/datafusion/).
 
 To get started, add the following to your `Cargo.toml` file:
@@ -27,3 +29,32 @@ To get started, add the following to your `Cargo.toml` file:
 [dependencies]
 datafusion = "5.1.0"
 ```
+
+## Optimized Configuration
+
+For an optimized build several steps are required. First, use the following in your `Cargo.toml`:
+
+```toml
+[dependencies]
+datafusion = { git = "https://github.com/apache/arrow-datafusion.git", features = ["simd"]}
+tokio = { version = "^1.0", features = ["macros", "rt", "rt-multi-thread"] }
+snmalloc-rs = {version = "0.2", features= ["cache-friendly"]}
+num_cpus = "1.0"
+
+[profile.release]
+lto = true
+codegen-units = 1
+```
+
+Then, in your `main.rs.` update the memory allocator with the below after your imports:
+
+```rust
+#[global_allocator]
+static ALLOC: snmalloc_rs::SnMalloc = snmalloc_rs::SnMalloc;
+```
+
+Finally, in order to build with these optimizations `cargo nightly` is required. Based on the instruction
+set architecture you are building on you will want to configure the `target-cpu` as well, ideally
+with `native` or at least `avx2`.
+
+`RUSTFLAGS='-C target-cpu=native' cargo +nightly run --release`

--- a/docs/user-guide/src/library.md
+++ b/docs/user-guide/src/library.md
@@ -32,11 +32,12 @@ datafusion = "5.1.0"
 
 ## Optimized Configuration
 
-For an optimized build several steps are required. First, use the following in your `Cargo.toml`:
+For an optimized build several steps are required. First, use the below in your `Cargo.toml`. It is
+worth noting that using the settings in the `[profile.release]` section will significantly increase the build time.
 
 ```toml
 [dependencies]
-datafusion = { git = "https://github.com/apache/arrow-datafusion.git", features = ["simd"]}
+datafusion = { version = "5.0" , features = ["simd"]}
 tokio = { version = "^1.0", features = ["macros", "rt", "rt-multi-thread"] }
 snmalloc-rs = {version = "0.2", features= ["cache-friendly"]}
 num_cpus = "1.0"
@@ -46,14 +47,14 @@ lto = true
 codegen-units = 1
 ```
 
-Then, in your `main.rs.` update the memory allocator with the below after your imports:
+Then, in `main.rs.` update the memory allocator with the below after your imports:
 
 ```rust
 #[global_allocator]
 static ALLOC: snmalloc_rs::SnMalloc = snmalloc_rs::SnMalloc;
 ```
 
-Finally, in order to build with these optimizations `cargo nightly` is required. Based on the instruction
+Finally, in order to build with the `simd` optimization `cargo nightly` is required. Based on the instruction
 set architecture you are building on you will want to configure the `target-cpu` as well, ideally
 with `native` or at least `avx2`.
 

